src/pages/index.js

```diff
@@ -1,140 +1,250 @@
+        card.handleIsLiked(true);
+      })
+      .catch((err) => {
+        console.log(`Unable to process request, ${err}`);
+      });
+  }
+}
 
-// Event Handlers
-function fillEditProfileForm() {
-  profileTitleInput.value = profileTitle.textContent;
-  profileDescriptionInput.value = profileDescription.textContent;
+function handleChangeProfilePicture(inputUrl) {
+  profileImageModal.renderLoading("Loading...");
+  api
+    .updateProfileImage(inputUrl)
+    .then((res) => {
+      userInfo.setProfileImage(res);
+      profileImageModal.close();
+      profileImageForm.resetValidation();
+      profileImageValidator.disableSubmit();
+    })
+    .catch((err) => {
+      console.log(`Unable to process request, ${err}`);
+    })
+    .finally(() => {
+      profileImageModal.renderLoading("Save");
+    });
 }
 
-// Event Listeners
+//ProfileEditButtonEventListeners
 profileEditButton.addEventListener("click", () => {
   profileEditValidator.hideInputError(profileEditForm);
-  fillEditProfileForm();
-  profileEditPopup.open();
+  const { name, about } = userInfo.getUserInfo();
+  profileTitleInput.value = name;
+  profileDescriptionInput.value = about;
+  profileEditModal.open();
+  profileEditValidator.resetValidation();
 });
-
+//ProfileEditImageEventListeners
+profileEditImage.addEventListener("click", () => {
+  profileEditValidator.hideInputError(profileImageForm);
+  const { url } = userInfo.getUserInfo();
+});
+//Add Card Button Event Listeners
 addNewCardButton.addEventListener("click", () => {
   addCardPopup.open();
+  addCardValidator.resetValidation();
 });
 
-profileEditForm.addEventListener("submit", (e) => {
-  e.preventDefault();
-  profileEditPopup._handleFormSubmit(profileEditForm);
+profileImageEditButton.addEventListener("click", () => {
+  profileImageModal.open();
 });
 
-addCardFormElement.addEventListener("submit", (e) => {
-  e.preventDefault();
-  const name = cardTitleInput.value;
-  const link = cardUrlInput.value;
-  renderCard({ name, link });
-  addCardPopup.close();
-  e.target.reset();
-  addCardValidator.toggleButtonState();
-});
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setProfileImage(userData);
+    userInfo.setUserInfo({
+      name: userData.name,
+      about: userData.about,
+    });
+  })
+  .catch((err) => {
+    console.log(`Unable to process request, ${err}`);
+  });
+
+api
+  .getInitialCards()
+  .then((cards) => {
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: (cardData) => {
+          renderCard(cardData);
+        },
+      },
+      ".cards__list"
+    );
+    cardSection.renderItems(cards);
+  })
+  .catch((err) => {
+    console.log(`Unable to process request, ${err}`);
+  });
+
+// Form Validators
+const profileImageValidator = new FormValidator(config, profileImageForm);
+const profileEditValidator = new FormValidator(config, profileEditForm);
+const addCardValidator = new FormValidator(config, addCardFormElement);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```